validations/transfer.test.ts

```diff
@@ -0,0 +1,136 @@
+import { expect, use } from 'chai'
+import { Wallet, ethers } from 'ethers'
+import path from 'path'
+import { randomUUID } from 'crypto'
+import { AddKycParams, FiatConnectClient } from '@fiatconnect/fiatconnect-sdk'
+import {
+  KycSchema,
+  TransferStatus,
+  Network,
+} from '@fiatconnect/fiatconnect-types'
+import { config } from '../src/config'
+import { chaiPlugin } from 'api-contract-validator'
+import { MOCK_QUOTE } from '../src/mock-data/quote'
+import { MOCK_KYC } from '../src/mock-data/kyc'
+import { MOCK_FIAT_ACCOUNTS } from '../src/mock-data/fiat-account'
+
+const apiDefinitionsPath = path.join(config.openapiSpec)
+use(chaiPlugin({ apiDefinitionsPath }))
+
+describe('/transfer', () => {
+  const mockAccountData =
+    MOCK_FIAT_ACCOUNTS[
+    config.fiatAccountMock as keyof typeof MOCK_FIAT_ACCOUNTS
+  ]
+
+  const mockKYCInfo: AddKycParams<KycSchema> =
```

Comment:
I know this is done elsewhere in the repo but I just realized it's sort of an abuse of the type argument-- the goal was for the argument to give you information on what the shape of the `data` field should be. The type is defined as follows:
```
export interface AddKycParams<T extends KycSchema> {
  kycSchemaName: T
  data: KycSchemas[T]
}
```
That being said, I'm not sure how to improve on this. We know from the kycMock object what kind of KYC is used here, so that should be usable (or maybe with slight modification to the kyc mock object). But after tinkering around for a few I didn't arrive at a solution I actually liked.

Reply:
@satish-ravi do you want to weigh in here? IIRC you created this type, is there a solution here I'm not seeing?

Reply:
I also spent a bit of time trying to get this work, but wasn't able to come up with anything that the compiler didn't yell at..

Reply:
I am not sure if there's a great solution here  since the type of `MOCK_KYC` has to be something generic since it can have object values of different KycSchemas. And the exact KycSchema here is decided at run time.

Reply:
One thing we could do is just drop the type for `mockKYCInfo` and `MOCK_KYC`. It doesn't make much difference for KYC right now because there's only one schema. But it makes a difference for `MOCK_FIAT_ACCOUNTS` because right now the type of `mockAccountData.data` could be any of the 5 fiat account schemas, but if we drop the `Record<string, PostFiatAccountRequestBody<FiatAccountSchema>>`, the type for `mockAccountData.data` would be `AccountNumber | IBANNumber`